ui/js/dfv/src/hooks/useBlockEditor.js

```diff
@@ -0,0 +1,77 @@
+
+const useBlockEditor = () => {
+	const editorSelect = wp.data?.select( 'core/editor' );
+	const editorDispatch = wp.data?.dispatch( 'core/editor' );
+	const notices = wp.data?.dispatch( 'core/notices' );
+
+	// @todo Use hook instead of savePost override once stable.
+	if ( ! window.PodsBlockEditor ) {
+		// First init.
+		window.PodsBlockEditor = {
+			// Store original.
+			savePost: editorDispatch.savePost,
+			messages: {},
+			callbacks: {},
+		};
+
+		// Override the current editor savePost function.
+		editorDispatch.savePost = async ( options ) => {
+			options = options || {};
+
+			const pbe = window.PodsBlockEditor;
+
+			if ( ! Object.values( pbe.messages ).length ) {
+				// eslint-disable-next-line no-undef
+				return pbe.savePost.apply( this, arguments );
+			}
+
+			return new Promise( function( resolve, reject ) {
+				// Bail early if is autosave or preview.
```

Comment:
@sc0ttkclark Should we allow saving for a draft?

Reply:
I'm on the fence here. Maybe we can make that an option in settings or a constant/hook. Default to allow saves on drafts perhaps.

Reply:
But there's no way to draft by default when you have a post that's already published. So we'd need to distinguish between save and save draft.